### PR TITLE
dev-python/webpy: new package

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**
web.py is a web framework for python that is as simple as it is powerful

Signed-off-by: Alessandro Barbieri <lssndrbarbieri@gmail.com>

### CODE REVIEW

**Comment by Alessandro-Barbieri** [comment]
merge https://github.com/gentoo/gentoo/pull/14980 before this

File: dev-python/webpy/webpy-0.40.ebuild [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="a web framework for python that is as simple as it is powerful"
+HOMEPAGE="http://www.webpy.org"
```
Comment (charIes17) [inline dev-python/webpy/webpy-0.40.ebuild]: HOMEPAGE="https://webpy.org"

File: dev-python/webpy/webpy-0.40.ebuild [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="a web framework for python that is as simple as it is powerful"
+HOMEPAGE="https://www.webpy.org"
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild]: Also add the Github page as 2ndary HOMEPAGE.

File: dev-python/webpy/webpy-0.40.ebuild [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# Copyright 1999-2017 Gentoo Foundation
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild]: Old, you should use repoman to commit so it fixes this for you automatically.

File: dev-python/webpy/webpy-0.40.ebuild [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="a web framework for python that is as simple as it is powerful"
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild]: Start with capital a, `A web frame...`

File: dev-python/webpy/webpy-0.40.ebuild [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+RDEPEND="
+	>=dev-python/cheroot-6.0.0[${PYTHON_USEDEP}]
+"
+DEPEND="
+	${RDEPEND}
+	test? (
+		dev-python/dbutils[${PYTHON_USEDEP}]
+		>=dev-python/mysql-connector-python-8.0.19[${PYTHON_USEDEP}]
+		>=dev-python/psycopg-2.8.4[${PYTHON_USEDEP}]
+		>=dev-python/pymysql-0.9.3[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild]: This version restriction can be lifted off as it's always satisfied in ::gentoo.

File: dev-python/webpy/webpy-0.40.ebuild [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+HOMEPAGE="https://www.webpy.org"
+SRC_URI="https://github.com/${PN}/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="public-domain"
+SLOT="0"
+KEYWORDS="~amd64"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-python/cheroot-6.0.0[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild]: This too.

File: dev-python/webpy/webpy-0.40.ebuild (hunk 3 above)
Comment (Alessandro-Barbieri) [reply dev-python/webpy/webpy-0.40.ebuild]: Can't use repoman for now because of broken dependencies

**Comment by Alessandro-Barbieri** [comment]
https://github.com/cherrypy/cheroot/blob/1959f3d9b310993f50935322b6e90cfe7a77e022/setup.cfg#L69

**Comment by Alessandro-Barbieri** [comment]
merge https://github.com/gentoo/gentoo/pull/15023 before this

**Comment by Alessandro-Barbieri** [comment]
@juippis tests works on my end

File: dev-python/webpy/webpy-0.50.ebuild [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+DEPEND="
+	${RDEPEND}
+	test? (
+		>=dev-python/pytest-4.6.2[${PYTHON_USEDEP}]
+	)
+"
+#		dev-python/dbutils[${PYTHON_USEDEP}]
+#		>=dev-python/mysql-connector-python-8.0.19[${PYTHON_USEDEP}]
+#		dev-python/pymysql[${PYTHON_USEDEP}]
+#		>=dev-python/psycopg-2.8.4[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/webpy/webpy-0.50.ebuild:34]: Can these be removed? What's their purpose?

File: dev-python/webpy/webpy-0.40.ebuild [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+DEPEND="
+	${RDEPEND}
+	test? (
+		>=dev-python/pytest-4.6.2[${PYTHON_USEDEP}]
+	)
+"
+#		dev-python/dbutils[${PYTHON_USEDEP}]
+#		>=dev-python/mysql-connector-python-8.0.19[${PYTHON_USEDEP}]
+#		dev-python/pymysql[${PYTHON_USEDEP}]
+#		>=dev-python/psycopg-2.8.4[${PYTHON_USEDEP}]
```
Comment (juippis) [inline dev-python/webpy/webpy-0.40.ebuild:34]: These too.

**Pull request closed without merging**